### ISSUE TITLE
Fix request: false positive in IUOC03 OptimizeMapOrSetForEnum

### ISSUE THREAD

**Issue by driekeerJ** [jborgers/PMD-jPinpoint-rules#162]
The following will give the error code: OptimizeMapOrSetForEnumMap has enum keys or Set has enum elemens, use EnumMap or EnumSet for more efficiency.

But offcourse it is already an EnumMap
````
final EnumMap<EnumTypeCode, List<SpecificModel>> specificModelsPerType = enums.stream()
        .collect(groupingBy(SpecificModel::getEnumTypeCode,
                () -> new EnumMap<>(EnumTypeCode.class),
                toList()));
````
